File: client/src/utils/trustLabel.ts
/**
 * P2-M7-06 · 分析可信度 tier 与文案（TrustBadge / 海报共用）
 */

export type TrustTier = 'high' | 'medium' | 'low'

export const HIGH_CONFIDENCE_THRESHOLD = 0.75
export const LOW_CONFIDENCE_THRESHOLD = 0.5

export function resolveTrustTier(confidence: number | null | undefined): TrustTier {
  const value = confidence ?? 1.0
  if (value >= HIGH_CONFIDENCE_THRESHOLD) return 'high'
  if (value >= LOW_CONFIDENCE_THRESHOLD) return 'medium'
  return 'low'
}

export function shouldRecommendRetake(confidence: number | null | undefined): boolean {
  const value = confidence ?? 1.0
  return value < LOW_CONFIDENCE_THRESHOLD
}

const TIER_COMPACT_TITLE: Record<TrustTier, string> = {
  high: 'AI 高可信',
  medium: 'AI 中等可信',
  low: 'AI 低可信',
}

/** 海报等窄区域用的单行标签（含百分比） */
export function formatTrustCompactLabel(confidence: number | null | undefined): string {
  const tier = resolveTrustTier(confidence)
  const pct = Math.round((confidence ?? 1.0) * 100)
  return `${TIER_COMPACT_TITLE[tier]} ${pct}%`
}

/**
 * P2-W11 · 历史列表卡片用的极短可信度标签（不含百分比）
 *
 * 卡片信息密度高，留给 trust 标签的横向空间只有几个字；带百分比反而干扰
 * "分数 vs 可信度"的视觉层级。所以列表里只显示「高/中/低」三档。
 */
const TIER_MINI_LABEL: Record<TrustTier, string> = {
  high: 'AI 高可信',
  medium: 'AI 中等可信',
  low: 'AI 低可信',
}
export function formatTrustMiniLabel(confidence: number | null | undefined): string {
  return TIER_MINI_LABEL[resolveTrustTier(confidence)]
}

File: app/lib/core/trust_label.dart
/// 可信度档位：对照 client/src/utils/trustLabel.ts。

enum TrustTier { high, medium, low }

const kHighConfidenceThreshold = 0.75;
const kLowConfidenceThreshold = 0.5;

TrustTier resolveTrustTier(num? confidence) {
  final value = (confidence ?? 1.0).toDouble();
  if (value >= kHighConfidenceThreshold) return TrustTier.high;
  if (value >= kLowConfidenceThreshold) return TrustTier.medium;
  return TrustTier.low;
}

bool shouldRecommendRetake(num? confidence) {
  final value = (confidence ?? 1.0).toDouble();
  return value < kLowConfidenceThreshold;
}

({String title, String hint}) trustTierCopy(TrustTier tier) => switch (tier) {
      TrustTier.high => (
          title: 'AI 高可信',
          hint: '画质、机位、姿态识别均良好，本次分析结果可信。',
        ),
      TrustTier.medium => (
          title: 'AI 中等可信，可参考',
          hint: '部分信号偏弱，建议在更好的光线/机位下再拍一段对比。',
        ),
      TrustTier.low => (
          title: 'AI 难以做出可靠分析',
          hint: '画质或机位影响识别，建议按提示重拍后再分析。',
        ),
    };
